libretroshare/src/pgp/rscertificate.h

```diff
@@ -54,7 +54,7 @@ class RsCertificate
 	bool initializeFromString(const std::string& str, uint32_t& errCode);
 
 	/// Constructs from binary gpg key, and RsPeerDetails.
-	RsCertificate( const RsPeerDetails& details,
+	RS_DEPRECATED RsCertificate( const RsPeerDetails& details,
```

Comment:
only deprecate if you have something in replacement. It's of course required to call that constructor in a try{}catch() block. Not very elegant but safe, and I'm not sure there is a proper way to recover from an error in the constructor. You'd need to add a isValid() member to the certificate object, etc. That's not so nice either.

Reply:
I forgot to add a function to replace that constructor like I did with the constructor from sting certificate, I am doing it right now ;)

Reply:
done